### PR TITLE
Update Checksums to support blacklisted taxonomies.

### PR DESCRIPTION
**Pull request by mdbitz**
The Checksum process rolled out in Jetpack 9.3 did not limit taxonomies based on existing blacklists. This update overhauls the table configuration and supporting functionality so that we don't identify content as missing that is not supposed to be Synced to the cache site.

#### Changes proposed in this Pull Request:
* Updates taxonomy checksums to exclude blacklisted taxonomies.

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:

* Perform a checksum of all tables.
```$rs = new Automattic\Jetpack\Sync\Replicastore();```
* Verify that no errors are returned and we have a checksum for all 7 tables 
* Note the checksum values for terms, term_taxonomy and term_relationships
* Add a new taxonomy of "ancestors", "comments_number", "the_categories" and "usermeta"
* Add terms to each of the new categories
* Add terms to existing posts
* Perform a new checksum of all tables.
* Verify that the checksum values for terms, term_taxonomy and term_relationships have not changed from before terms/taxonomies were added.

#### Proposed changelog entry for your changes:
* Jetpack Sync - Update checksum process to exclude blacklisted taxonomies.

### CODE REVIEW

**Comment by mdbitz** [comment]
oddly there is a failure on `test_change_owner` which would not be impacted by any changes. re-running tests.

File: projects/packages/sync/src/replicastore/class-table-checksum.php [hunk 1]
```diff
@@ -451,10 +488,10 @@ private function build_checksum_query( $range_from = null, $range_to = null, $fi
 		$join_statement = '';
 		if ( $this->parent_table ) {
 			$parent_table_obj    = new Table_Checksum( $this->parent_table );
-			$parent_filter_query = $parent_table_obj->build_filter_statement( $range_from, $range_to, null, 'parent_table' );
+			$parent_filter_query = $parent_table_obj->build_filter_statement( $range_from, $range_to, null, 'parent_table', true );
```
Comment (mdbitz) [inline projects/packages/sync/src/replicastore/class-table-checksum.php:491]: revert the addition of $exclude_ranges.

Instead simply pass null for $range_from and $range_to for the parent query..

File: projects/packages/sync/src/replicastore/class-table-checksum.php [hunk 2]
```diff
@@ -323,6 +351,7 @@ private function prepare_filter_values_as_sql( $filter_values = array(), $table_
 			switch ( $filter['operator'] ) {
 				case 'IN':
 				case 'NOT IN':
+				case '<>':
```
Comment (bisko) [inline projects/packages/sync/src/replicastore/class-table-checksum.php:354]: This won't work as expected.

Adding multiple arguments to a `<>` with this syntax will create the following query:

`SELECT * FROM wp_postmeta WHERE meta_key <> ('_wp_page_template', '_edit_lock')`

that will result in `ERROR 1241 (21000) at line 1: Operand should contain 1 column(s)`.

The proper syntax generated by `<>` should be:

```
column <> value AND column <> value2 AND column <> value3 
```

**Review by bisko — CHANGES_REQUESTED** [review]

**Comment by bisko** [comment]
Quick way of testing for the future. Could be used as a one-liner in `wp shell`.
```
$rs = new Automattic\Jetpack\Sync\Replicastore(); 
foreach ( [ 'posts', 'postmeta', 'comments', 'commentmeta', 'terms', 'termmeta', 'term_relationships', 'term_taxonomy' ] as $table) { 
    echo "$table\n"; 
    var_export($rs->checksum_histogram($table, 2, 10, 1000, null, true, '', false, false)); 
    echo "\n";
};
```

**Review by bisko — APPROVED** [review]

File: projects/packages/sync/src/replicastore/class-table-checksum.php (hunk 2 above)
Comment (mdbitz) [reply projects/packages/sync/src/replicastore/class-table-checksum.php]: :thumbsup: The <> was how the existing sql statement was as we were looking at a single status of 'spam'. No harm in switching to use NOT IN.

**Pull request merged**